src/overlays/actors/ovl_En_Rat/z_en_rat.c

```diff
@@ -28,123 +47,913 @@ const ActorInit En_Rat_InitVars = {
+    }
+
+    angle = func_80086C48(normDotUp);
+    if (angle < 0.001f) {
+        return false;
+    }
+
+    Math3D_CrossProduct(&this->axisUp, &normal, &vec);
+
+    magnitude = Math3D_Vec3fMagnitude(&vec);
+
+    if (magnitude < 0.001f) {
+        EnRat_Explode(this, play);
+        return false;
+    }
+
+    Math_Vec3f_Scale(&vec, 1.0f / magnitude);
+    Matrix_RotateAxisF(angle, &vec, MTXMODE_NEW);
+    Matrix_MultVec3f(&this->axisLeft, &vec);
+    Math_Vec3f_Copy(&this->axisLeft, &vec);
+    Math3D_CrossProduct(&this->axisLeft, &normal, &this->axisForwards);
+
+    magnitude = Math3D_Vec3fMagnitude(&this->axisForwards);
+    if (magnitude < 0.001f) {
+        EnRat_Explode(this, play);
+        return false;
+    }
+
+    Math_Vec3f_Scale(&this->axisForwards, 1.0f / magnitude);
+    Math_Vec3f_Copy(&this->axisUp, &normal);
+    return true;
+}
+
+void EnRat_UpdateRotation(EnRat* this) {
+    MtxF mf;
+
+    mf.xx = this->axisLeft.x;
+    mf.yx = this->axisLeft.y;
+    mf.zx = this->axisLeft.z;
+
+    mf.xy = this->axisUp.x;
+    mf.yy = this->axisUp.y;
+    mf.zy = this->axisUp.z;
+
+    mf.xz = this->axisForwards.x;
+    mf.yz = this->axisForwards.y;
+    mf.zz = this->axisForwards.z;
+
+    Matrix_MtxFToYXZRot(&mf, &this->actor.world.rot, false);
+    this->actor.world.rot.x = -this->actor.world.rot.x;
+}
+
+/**
+ * Chooses to either look at the player (if the Real Bombchu has spotted the player or
+ * is chasing after them) or at some other semi-random point (if the Bombchu is idle).
+ */
+void EnRat_ChooseDirection(EnRat* this) {
+    Vec3f sp74;
+    s16 angle;
+
+    if (this->actionFunc != EnRat_Idle) {
+        angle = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+        if (this->axisUp.y < -0.25f) {
+            angle -= 0x8000;
+        }
+    } else {
+        if (Actor_DistanceToPoint(&this->actor, &this->actor.home.pos) > 50.0f) {
+            Vec3f sp64;
+            Vec3f sp58;
+            Vec3f sp4C;
+            Vec3f sp40;
+
+            Matrix_RotateZS(-this->actor.home.rot.z, MTXMODE_NEW);
+            Matrix_RotateXS(-this->actor.home.rot.x, MTXMODE_APPLY);
+            Matrix_RotateYS(-this->actor.home.rot.y, MTXMODE_APPLY);
+            Matrix_MultVec3f(&this->axisUp, &sp40);
+            Math_Vec3f_Sum(&this->actor.world.pos, &this->axisForwards, &sp64);
+            Matrix_MultVec3f(&sp64, &sp4C);
+            Matrix_MultVec3f(&this->actor.home.pos, &sp64);
+            Matrix_MultVec3f(&this->actor.world.pos, &sp58);
+            angle = Math_Vec3f_Yaw(&sp58, &sp64) - Math_Vec3f_Yaw(&sp58, &sp4C);
+            if (sp40.y < -0.25f) {
+                angle -= 0x8000;
+            }
+
+            angle += (s16)randPlusMinusPoint5Scaled(2048.0f);
+        } else {
+            angle = (Rand_ZeroOne() < 0.1f) ? (s16)randPlusMinusPoint5Scaled(2048.0f) : 0;
+        }
+    }
+
+    angle = CLAMP(angle, -0x800, 0x800);
+    Matrix_RotateAxisF(angle * 0.0000958738f, &this->axisUp, MTXMODE_NEW);
```

Comment:
```suggestion
    Matrix_RotateAxisF(angle * (M_PI / 0x8000), &this->axisUp, MTXMODE_NEW);
```
We really should consider an `BINANG_TO_RAD_ALT` or something for cases like this. It'd be worth comparing the possible usage of that to the current `#define BINANG_TO_RAD(binang) (((f32)binang / 0x8000) * M_PI)`